arviz/data/io_pyro.py

```diff
@@ -45,10 +66,18 @@ def __init__(
             self.nchains = self.ndraws = 0
         self.prior = prior
         self.posterior_predictive = posterior_predictive
+        self.predictions = predictions
+        self.constant_data = constant_data
+        self.predictions_constant_data = predictions_constant_data
         self.coords = coords
         self.dims = dims
+        self.pred_dims = pred_dims
+        self.num_chains = num_chains
```

Comment:
I would handle this in the `else: self.nchains = self.ndraws = 0` below, to set self.nchains to num_chains and then get the number of draws from the predictions, posterior predictive or prior (I think io_pymc does something similar).

Reply:
Also, as it is only used in init, it does not need to be saved in self